Documentation/proposals/202501-configuration-object-status-subresource.md

```diff
@@ -0,0 +1,322 @@
+
+## Goals
+
+* Define the structure of the status subresource for the custom resource definitions
+  * `ServiceMonitor`
+  * `PodMonitor`
+  * `ScrapeConfig`
+  * `Probes`
+  * `PrometheusRule`
+  * `AlertmanagerConfig`
+* Provide information about the targets being scraped and their status for scrape resources (`PodMonitor`, `ServiceMonitor`, `Probes` and `ScrapeConfig`).
+* Report when a configuration resource is considered invalid during reconciliation. For example:
+  * Feature not being supported by the version of the workload.
+  * Invalid configmap/secret key reference.
+  * Invalid PromQL expression in PrometheusRule resources.
+
+## Non-Goals
+
+* The solution does not aim to expose the full live configuration or runtime status of Prometheus. Doing so would be expensive for the workload, the operator and the Kubernetes system in general.
+  * It will not include per-target scrape status, only summary information (e.g., number of targets up/down).
+  * It will not surface fired alerts from PrometheusRule resources, as querying Prometheus for this data can be expensive and places undue load on Prometheus, the operator and the Kubernetes platform in general.
+* Configuration resources won't expose status information that explains why they are not being selected by Prometheus or why their targets are not being scraped.
+  * This non-goal can be partially addressed by tools like [`poctl`](https://github.com/prometheus-operator/poctl), which provide insights into configuration resource selection and target matching.
+
+### Audience
+
+* Users of Prometheus-Operator
+* Maintainers and Contributors of Prometheus-Operator
+
+## How
+
+There are different challenges that influence the API design:
+
+* A single config resource can be selected by multiple workload resources.
+* The configuration resource may not be in the same namespace as the workload resource.
+* Which workload selects which configuration resources can vary over time depending on the workload resource's label selectors and on the configuration resource's labels.
+
+### CRDs
+
+#### `ServiceMonitor`/`PodMonitor`/`Probes`/`ScrapeConfig`
+
+```yaml
+apiVersion: monitoring.coreos.com/v1
+kind: ServiceMonitor
+metadata:
+  name: example-servicemonitor
+  namespace: monitoring
+  generation: 3
+  labels:
+    team: backend
+spec:
+  selector:
+    matchLabels:
+      app: my-service
+  namespaceSelector:
+    matchNames:
+      - default
+  endpoints:
+    - port: http
+      path: /metrics
+      interval: 30s
+      scheme: http
+      basicAuth:
+        username:
+          name: my-secret
+          key: basic-auth-username
+        password:
+          name: my-secret
+          key: basic-auth-password
+  status:
+    bindings:
+      - group: monitoring.coreos.com
```

Comment:
(nit) for clarity we could nest ref fields in a dedicated field.
```suggestion
      - workloadRef:
          group: group: monitoring.coreos.com
          resource: ...
          name: ...
          namespace: ...
```

Reply:
I wouldn't be worried about the additional nesting.
But after more investigation, it isn't compatible with `+listType=map` and `+listMapKey` (the latter supports only scalar fields).